packages/flutter/test/painting/text_painter_test.dart

```diff
@@ -996,6 +996,18 @@ void main() {
         ui.Rect.zero);
     expect(caretOffset.dx, painter.width);
   }, skip: kIsWeb && !isCanvasKit); // https://github.com/flutter/flutter/issues/87545
+
+  test('TextPainter paragraph is null test', () {
+    final TextPainter painter = TextPainter()
+      ..textDirection = TextDirection.ltr;
+
+    const String text = 'A';
+    painter.text = const TextSpan(text: text, style: TextStyle(height: 1.0));
+    painter.layout();
+
+    painter.markNeedsLayout();
```

Comment:
I think the use case tested here is invalid. When `painter.markNeedsLayout` is called you're not supposed to be able to call `painter.getPositionForOffset`, and the program probably should throw to indicate the programmer error.

Reply:
If a render object's layout information is out of date, it shouldn't be hit tested in the first place I think.

Reply:
Hi, in the previous code, `_paragraph` is null only in two stages, its initialization and before `layout` after `markNeedsLayout`, i think #84650 happened just in this situation. 
In my opinion, `markNeedsLayout()` does works in the next pipeline. Maybe in current pipeline we still needs its text information. @LongCatIsLooong

Reply:
Hit testing is done in the `idle` phase I think. @goderbauer is it possible for other platform channel messages to arrive before the touch events in that frame, and invalidate the layout before hit testing?

Reply:
@LongCatIsLooong in the online environment, this pr does works. before `layout` and after `markNeedsLayout` `hitTest` is called i think.

Reply:
I think my point still stands: if a text painter is dirty it probably shouldn't be hit-tested. For instance, the text in the text painter could be different, so I think ideally we would want to prevent the framework from calling `hitTest` without `layout` being called on the text painter, instead of keep using the outdated text layout cache.

Reply:
@LongCatIsLooong I agree with you, it's better to prevent the framework to do this. But for the defensive programming, this change still necessary i think. ( it's hard to know how the framework do it from the current calling stack. So maybe I need to modify the test?